src/parser.rs

```diff
@@ -153,24 +145,48 @@ pub fn variable_name(input: &str) -> IResult<&str, String, Error<&str>> {
     map(alpha1, |x: &str| x.to_string())(input)
 }
 
-pub fn assignment(input: &str) -> IResult<&str, Expression, Error<&str>> {
+pub fn assignment(input: &str) -> IResult<&str, Command, Error<&str>> {
     map(
         tuple((variable_name, space0, char(':'), space0, expr)),
-        |(variable_name, _, _, _, value)| Expression::Declaration((variable_name, value)),
+        |(variable_name, _, _, _, value)| Command::Declaration((variable_name, value)),
     )(input)
 }
 
-pub fn node_initialization(input: &str) -> IResult<&str, Expression, Error<&str>> {
-    map(alt((tag("square"), tag("box"))), |node| match node {
-        "square" => Expression::Instantiation(Node::Square((1.0, 1.0))),
-        "circle" => Expression::Instantiation(Node::Circle(1.0)),
-        _ => unreachable!(),
-    })(input)
+pub fn draw_shape(input: &str) -> IResult<&str, Command, Error<&str>> {
+    alt((
+        map(
```

Comment:
Writing separated parsers, you can use `many_m_n` to have optional params, something like this:
```rust
pub fn square(input: &str) -> IResult<&str, Node, Error<&str>> {
    map(
        pair(tag("square"), many_m_n(0, 2, expr)),
        |(_, params)| match (params.get(0), params.get(1)) {
            (Some(width), Some(height)) => Node::Square((width.clone(), height.clone())),
            (Some(size), None) => Node::Square((size.clone(), size.clone())),
            (None, None) => Node::Square((
                Operation::Identity(Factor::Number(1.0)),
                Operation::Identity(Factor::Number(1.0)),
            )),
            _ => unreachable!(),
        },
    )(input)
}

pub fn circle(input: &str) -> IResult<&str, Node, Error<&str>> {
    map(
        pair(tag("circle"), many_m_n(0, 1, expr)),
        |(_, params)| match params.first() {
            Some(radius) => Node::Circle(radius.clone()),
            None => Node::Circle(Operation::Identity(Factor::Number(1.0))),
        },
    )(input)
}


```